WordPress/Classes/ViewRelated/Blog/BlogListViewController.m

```diff
@@ -599,6 +608,47 @@ - (void)visibilitySwitchAction:(id)sender
+            self.firstDisable = [NSDate date];
+        }
+        self.disabledCount += 1;
+
+        if (self.disabledCount > DisableAllSitesThreshold && (self.firstDisable.timeIntervalSinceNow * -1) < DisableAllSitesInterval) {
+            
+            NSString *message = NSLocalizedString(@"Would you like to hide all WordPress.com Sites?",
+                                                  @"Message offering to hide all WPCom Sites");
+
+            UIAlertController *alertController = [UIAlertController alertControllerWithTitle:NSLocalizedString(@"Hide All Sites", @"Hide All Sites")
+                                                                                     message:message
+                                                                              preferredStyle:UIAlertControllerStyleAlert];
+            
+            UIAlertAction *cancelAction = [UIAlertAction actionWithTitle:NSLocalizedString(@"Cancel", @"Cancel")
+                                                                   style:UIAlertActionStyleCancel
+                                                                 handler:^(UIAlertAction *action){}];
+            
+            UIAlertAction *deleteAction = [UIAlertAction actionWithTitle:NSLocalizedString(@"Hide All", @"Hide All")
+                                                                   style:UIAlertActionStyleDestructive
+                                                                 handler:^(UIAlertAction *action){
+                                                                     NSManagedObjectContext *context = [[ContextManager sharedInstance] newDerivedContext];
+                                                                     [context performBlock:^{
+                                                                         BlogService *blogService = [[BlogService alloc] initWithManagedObjectContext:context];
+                                                                         NSArray *blogs = [blogService blogsWithPredicate:[self fetchRequestPredicateForHideableBlogs]];
+                                                                         
+                                                                         if(blogs == nil) {
+                                                                             return;
+                                                                         }
+                                                                         
+                                                                         AccountService *accountService = [[AccountService alloc] initWithManagedObjectContext:context];
+                                                                         [accountService setVisibility:switcher.on forBlogs:blogs];
+                                                                     }];
+                                                                     [[ContextManager sharedInstance] saveDerivedContext:context];
```

Comment:
The call to `saveDerivedContext` should probably be made from inside the `performBlock`.  Even though the way queuing works means it *shouldn't* be called before the performBlock is executed, it could be problematic if our implementation of `saveDerivedContext` changes in the future.  The intent of the code is also a tad clearer when its inside the `performBlock`.